src/basic/env-util.c

```diff
@@ -336,7 +349,7 @@ char **strv_env_unset(char **l, const char *p) {
 
         for (f = t = l; *f; f++) {
 
-                if (env_match(*f, p)) {
+                if (env_entry_has_name(*f, p)) {
```

Comment:
is this change really the best idea? Previously, you could call `strv_env_delete()` with FOO=BAR and it would remove exactly FOO=BAR, but never FOO=QUUX. You could also call it with just FOO in which case FOO=BAR and FOO=QUUX are removed.

I think this behaviour makes a lot of sense, and we do expose it in our bus calls for seting the env vars, afaics. THis chnage doesn't look right to me hence?